src/Base/Unit.h

```diff
@@ -1,179 +1,260 @@
+    { "MagneticFluxDensity"                   , {  0,  1, -2, -1                 } },
+    { "Magnetization"                         , { -1,  0,  0,  1                 } },
+    { "Moment"                                , {  2,  1, -2                     } },
+    { "Pressure"                              , { -1,  1, -2                     } },
+    { "Power"                                 , {  2,  1, -3                     } },
+    { "ShearModulus"                          , { -1,  1, -2                     } },
+    { "SpecificEnergy"                        , {  2,  0, -2                     } },
+    { "SpecificHeat"                          , {  2,  0, -2,  0, -1             } },
+    { "Stiffness"                             , {  0,  1, -2                     } },
+    { "StiffnessDensity"                      , { -2,  1, -2                     } },
+    { "Stress"                                , { -1,  1, -2                     } },
+    { "SurfaceChargeDensity"                  , { -2,  0,  1,  1                 } },
+    { "ThermalConductivity"                   , {  1,  1, -3,  0, -1             } },
+    { "ThermalExpansionCoefficient"           , {  0,  0,  0,  0, -1             } },
+    { "ThermalTransferCoefficient"            , {  0,  1, -3,  0, -1             } },
+    { "UltimateTensileStrength"               , { -1,  1, -2                     } },
+    { "VacuumPermittivity"                    , { -3, -1,  4,  2                 } },
+    { "Velocity"                              , {  1,  0, -1                     } },
+    { "Volume"                                , {  3                             } },
+    { "VolumeChargeDensity"                   , { -3,  0,  1,  1                 } },
+    { "VolumeFlowRate"                        , {  3,  0, -1                     } },
+    { "VolumetricThermalExpansionCoefficient" , {  0,  0,  0,  0, -1             } },
+    { "Work"                                  , {  2,  1, -2                     } },
+    { "YieldStrength"                         , { -1,  1, -2                     } },
+    { "YoungsModulus"                         , { -1,  1, -2                     } },
+}};  // clang-format on
+
+inline bool isUnitName(const std::string_view str)
 {
-    *this = *this * that;
-    return *this;
+    auto match = [&](const std::pair<std::string_view, UnitVals>& pair) {
+        return pair.first == str;
+    };
+
+    const auto found = std::find_if(unitSpecs.begin(), unitSpecs.end(), match);
+    return found != unitSpecs.end();
 }
 
-inline Unit& Unit::operator/=(const Unit& that)
+inline std::string getString(const UnitVals vals)
 {
-    *this = *this / that;
-    return *this;
+    auto match = [&](const std::pair<std::string_view, UnitVals>& pair) {
+        return pair.second == vals;
+    };
+
+    const auto found = std::find_if(unitSpecs.begin(), unitSpecs.end(), match);
+    return std::string(found == unitSpecs.end() ? "" : found->first);
```

Comment:
I would suggest moving this to .cpp, to not have `#include <algorithm>" in the whole codebase

Reply:
Right, this specific function uses `algorithm` so it makes sense to drop `inline` and move the implementation to the cpp file.